### PR TITLE
Type annotations

### PR DESCRIPTION
**Pull request by jorgelainfiesta**
This PR implements several type annotations around the app. Events are left without annotations for the moment.

### CODE REVIEW

File: src/ui/components/Breethe/component.ts [hunk 1]
```diff
@@ -97,12 +103,11 @@ export default class Breethe extends Component {
     if (!this.loadedLocal) {
       let transform = await this.local.pull((q) => q.findRecords());
       await this.store.sync(transform);
-      await this.coordinator.activate();
```
Comment (marcoow) [inline src/ui/components/Breethe/component.ts]: Is this not necessary? According to http://orbitjs.com/v0.15/guide/getting-started.html#Introducing-a-coordinator it should be? Anyway, I think the change is unrelated to the purpose of this PR.

File: tslint.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
     "only-arrow-functions": false,
     "prefer-const": false,
     "member-access": false,
-    "object-literal-sort-keys": false
+    "object-literal-sort-keys": false,
+    "interface-name": [true, "never-prefix"]
```
Comment (marcoow) [inline tslint.json]: Why do we opt out of prefixing interfaces?

Comment (jorgelainfiesta) [reply tslint.json]: Glimmer doesn't use this convention when defining its interfaces. I thought it'd be better to be consistent with them.

File: src/ui/components/Breethe/component.ts (hunk 1 above)
Comment (jorgelainfiesta) [reply src/ui/components/Breethe/component.ts]: Right, it's not related to this PR. I can remove it, but I found out it's redundant with https://github.com/simplabs/breethe-client/blob/type-annotations/src/utils/data/setup-store.ts#L68. The coordinator is already activated by the time it reaches this line.

File: tslint.json (hunk 2 above)
Comment (marcoow) [reply tslint.json]: yep, that makes sense 👍

**Comment by jorgelainfiesta** [comment]
I'll merge now that both observations have been addressed.
